docs/reference/sql/data-definition.md

```diff
@@ -0,0 +1,228 @@
+JSON format.
+
+Under the hood, each stream corresponds to a [Kafka topic](../../../overview/apache-kafka-primer/#topics)
+with a registered schema. If the backing topic for a stream doesn't exist when
+you declare it, ksqlDB creates it on your behalf, as shown in the previous
+example statement.
+
+You can also declare a stream on top of an existing topic. When you do that,
+ksqlDB simply registers its associated schema. If topic `s2` already exists,
+the following statement register a new stream over it:
+
+```sql
+CREATE STREAM s2 (
+    k1 VARCHAR KEY,
+    v1 VARCHAR
+) WITH (
+    kafka_topic = 's2',
+    value_format = 'json'
+);
+```
+
+!!! tip
+    When you create a stream on an existing topic, you don't need to declare
+    the number of partitions for the topic. ksqlDB infers the partition count
+    from the existing topic.
+
+## Tables
+
+A table is a mutable, partitioned collection that models change over time. In
+contrast with a stream, which represents a historical sequence of events, a
+table represents what is true as of "now". For example, you might use a table
+to model the locations where someone has lived as a stream: first Miami, then
+New York, then London, and so forth.
+
+Tables work by leveraging the keys of each row. If a sequence of rows shares a
+key, the last row for a given key represents the most up-to-date information
+for that key's identity. A background process periodically runs and deletes all
+but the newest rows for each key.
+
+Syntactically, declaring a table is similar to declaring a stream. The following
+example statement declares a `current_location` table that has a key field 
+named `person`.
+
+```sql
+CREATE TABLE current_location (
+    person VARCHAR PRIMARY KEY,
+    location VARCHAR
+) WITH (
+    kafka_topic = 'current_location',
+    partitions = 3,
+    value_format = 'json'
+);
+```
+
+As with a stream, you can declare a table directly on top of an existing
+{{ site.ak }} topic by omitting the number of partitions in the `WITH` clause.
+
+## Keys
+
+You can mark a column with the `KEY` keyword to indicate that it's a key
+column. Key columns constitute the key portion of the row's underlying
+{{ site.ak }} record. Only streams can mark columns as keys, and it's optional
+for them to do do. Tables must use the `PRIMARY KEY` constraint instead.
+
+In the following example statement, `k1`'s data is stored in the key portion of
+the row, and `v1`'s data is stored in the value.
+
+```sql
+CREATE STREAM s3 (
+    k1 VARCHAR KEY,
+    v1 VARCHAR
+) WITH (
+    kafka_topic = 's3',
+    value_format = 'json'
+);
+```
+
+The ability to declare key columns explicitly is especially useful when you're
+creating a stream over an existing topic. If ksqlDB can't infer what data is in
+the key of the underlying {{ site.ak }} record, it must perform a repartition
+of the rows internally. If you're not sure what data is in the key or you simply
+don't need it, you can omit the `KEY` keyword.
```

Comment:
> If you're not sure what data is in the key or you simply don't need it, you can omit the `KEY` keyword.

If find this misleading.  This suggests omitting just the `KEY` keyword is sufficient for ksqlDB to load the `k1` column in the above example correctly.  

```suggestion
of the rows internally. If you're not sure what data is in the key or you simply
don't need it, you can omit the `KEY` column. For example:

```sql
CREATE STREAM s3 (
    v1 VARCHAR
) WITH (
    kafka_topic = 's3',
    value_format = 'json'
);
```